File: app/src/main/java/org/stepic/droid/storage/StoreStateManager.java
package org.stepic.droid.storage;

import org.stepic.droid.model.Step;

public interface StoreStateManager {

    void updateUnitLessonState(long lessonId);

    void updateUnitLessonAfterDeleting(long lessonId);

    void updateStepAfterDeleting(Step step);

    void updateSectionAfterDeleting(long sectionId);

    void updateSectionState(long sectionId);
}

File: app/src/main/java/org/stepic/droid/storage/StoreStateManager.kt
package org.stepic.droid.storage

import org.stepik.android.model.structure.Step

//todo: do interface segregation: for units and for sections
interface StoreStateManager {

    fun updateUnitLessonState(lessonId: Long)

    fun updateUnitLessonAfterDeleting(lessonId: Long)

    fun updateStepAfterDeleting(step: Step?)

    fun updateSectionAfterDeleting(sectionId: Long)

    fun updateSectionState(sectionId: Long)


    interface LessonCallback {
        fun onLessonCached(lessonId: Long)

        fun onLessonNotCached(lessonId: Long)
    }

    fun addLessonCallback(callback: LessonCallback)

    fun removeLessonCallback(callback: LessonCallback)


    interface SectionCallback {
        fun onSectionCached(sectionId: Long)

        fun onSectionNotCached(sectionId: Long)
    }

    fun addSectionCallback(callback: SectionCallback)

    fun removeSectionCallback(callback: SectionCallback)

}
